m4/ax_check_pam.m4

```diff
@@ -26,34 +26,28 @@
 AC_DEFUN([AX_CHECK_PAM], [
     AC_ARG_WITH([pam],
         [AS_HELP_STRING([--with-pam=DIR],
-            [With libpam support, see ftp.redhat.com:/pub/pam])],
-        [
-            case "$withval" in
-            "" | y | ye | yes)
-                pamdirs="/usr/local /usr/lib /usr"  
-              ;;
-            n | no)
-                with_pam="no"
-              ;;
-            *)
-                pamdirs="$withval"
-              ;;
-            esac
-        ])
+            [With libpam support, see ftp.redhat.com:/pub/pam])])
     
-    if [ test "x${with_pam}" != "xno" ] ; then
-        PAM_LIBS="-lpam"
-        for pamdir in $pamdirs; do
-            AC_MSG_CHECKING([for pam_appl.h in $pamdir])
-            if test -f "$pamdir/include/security/pam_appl.h"; then
-                PAM_CFLAGS="-I$pamdir/include"
-                PAM_LDFLAGS="-L$pamdir/lib"
-                AC_MSG_RESULT([yes])
-                break
-            else
-                AC_MSG_RESULT([no])
-            fi
-        done
+    AS_IF([test "$with_pam" != "no"], [
```

Comment:
Generally, the convention is to use "x${with_pam}" != "xno", why change it?

Reply:
I have never understood the convention. Can you explain why the "x" is significant?

Reply:
It's a configure / shell compatibility thing. I've seen people insisting on putting it in to make these script consistent. You can look at the changelog for configure.ac

Reply:
I have restored the "x" when passing strings to the test command.